test/Mono.Linker.Tests.Cases/RequiresCapability/RequiresUnreferencedCodeCapability.cs

```diff
@@ -212,6 +210,8 @@ static void TestStaticCctorRequiresUnreferencedCode ()
 			_ = new StaticCtor ();
 		}
 
+		[ExpectedWarning ("IL2026", "'Mono.Linker.Tests.Cases.RequiresCapability.RequiresUnreferencedCodeCapability.DynamicallyAccessedTypeWithRequiresUnreferencedCode.RequiresUnreferencedCode()' method " +
+			"has 'RequiresUnreferencedCodeAttribute' which can break functionality when trimming application code. Message for --RequiresUnreferencedCode--.")]
 		public class DynamicallyAccessedTypeWithRequiresUnreferencedCode
```

Comment:
I don't understand why the warning should be reported on the type. The type itself does not have a reference - it's not the "usage". The usage in this case is in the Main method. So ideally the warning should be reported there.

Good way to think about this:
* Let's say the type is DangerousFeature and has public .ctor which is marked with RequiresUnreferencedCode - from assembly Dangerous.dll
* My app has a package reference to Dangerous
* In my app I use `Lazy<DangerousFeature>` in my Main (Lazy<T> has annotation on T to preserve public ctor)
* Where should the warning go?
  * With the current code it would be reported on DangerousFeature - but that makes little sense, that thing is annotated as dangerous, so why add a warning to it
  * The place which causes the Dangerous to be included is in the Main, so the warning should be reported there.
  * It also doesn't belong to Lazy - that in itself didn't do anything which would make the app problematic for trimming.

Reply:
Got it. I was a little confused by a previous comment, I've changed this back.